module/zfs/zvol.c

```diff
@@ -166,8 +166,11 @@ zvol_find_by_dev(dev_t dev)
 	ASSERT(MUTEX_HELD(&zvol_state_lock));
 	for (zv = list_head(&zvol_state_list); zv != NULL;
 	    zv = list_next(&zvol_state_list, zv)) {
+		mutex_enter(&zv->zv_state_lock);
 		if (zv->zv_dev == dev)
+			/* return with zv_state_lock taken */
```

Comment:
nit: I'd suggest we move this comment in to the block comment above the function and clearly describe the expected behavior of returning with the lock held when a match is found.  The same for `zvol_find_by_name_hash` and `zvol_find_by_name`.